### PR TITLE
HIVE-27783: Iceberg: Implement Copy-On-Write for Update queries

### PR DESCRIPTION
**Pull request by deniskuzZ**
### What changes were proposed in this pull request?

COW support for update operation

### Why are the changes needed?

part of COW support

### Does this PR introduce _any_ user-facing change?

No

### Is the change a dependency upgrade?

No

### How was this patch tested?

mvn clean test -Dtest=TestIcebergCliDriver -Dqfile=update_iceberg_copy_on_write_partitioned.q
mvn clean test -Dtest=TestIcebergCliDriver -Dqfile=update_iceberg_copy_on_write_unpartitioned.q

depends on #4700

### CODE REVIEW

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java [hunk 1]
```diff
@@ -1590,10 +1593,18 @@ public Map<String, String> getNativeProperties(org.apache.hadoop.hive.ql.metadat
   public boolean shouldOverwrite(org.apache.hadoop.hive.ql.metadata.Table mTable, String operationName) {
     String mode = null;
     String formatVersion = mTable.getTTable().getParameters().get(TableProperties.FORMAT_VERSION);
-    // As of now only delete mode is supported, for all others return false
-    if ("2".equals(formatVersion) && operationName.equalsIgnoreCase(Context.Operation.DELETE.toString())) {
-      mode = mTable.getTTable().getParameters()
-          .getOrDefault(TableProperties.DELETE_MODE, TableProperties.DELETE_MODE_DEFAULT);
+    // As of now only update & delete modes are supported, for all others return false
+    if ("2".equals(formatVersion)) {
+      switch (Context.Operation.valueOf(operationName)) {
```
Comment (kasakrisz) [inline iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:1598]: Ignore case is lost here. 
Should it be
```
Context.Operation.valueOf(operationName.toUpperCase())
```
?

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java [hunk 2]
```diff
@@ -1590,10 +1593,18 @@ public Map<String, String> getNativeProperties(org.apache.hadoop.hive.ql.metadat
   public boolean shouldOverwrite(org.apache.hadoop.hive.ql.metadata.Table mTable, String operationName) {
     String mode = null;
     String formatVersion = mTable.getTTable().getParameters().get(TableProperties.FORMAT_VERSION);
-    // As of now only delete mode is supported, for all others return false
-    if ("2".equals(formatVersion) && operationName.equalsIgnoreCase(Context.Operation.DELETE.toString())) {
-      mode = mTable.getTTable().getParameters()
-          .getOrDefault(TableProperties.DELETE_MODE, TableProperties.DELETE_MODE_DEFAULT);
+    // As of now only update & delete modes are supported, for all others return false
+    if ("2".equals(formatVersion)) {
```
Comment (zhangbutao) [inline iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:1597]: Acutually, both V1 and V2  support  copy on write. The current qtests are all limited to v2 iceberg tables. Maybe we can fix this after all the COW modes(merge/update/delete) are supported.

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java (hunk 1 above)
Comment (deniskuzZ) [reply iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java:1598]: omg, thanks!

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java (hunk 2 above)
Comment (deniskuzZ) [reply iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java]: 👍

**Review by kasakrisz — APPROVED** [review]
LGTM

**Pull request merged**